sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -1372,6 +1372,28 @@ class PlannerSuite extends SharedSparkSession with AdaptiveSparkPlanHelper {
       assert(numOutputPartitioning.size == 8)
     }
   }
+
+  test("SPARK-44804: SortMergeJoin should respect the streamed side ordering") {
```

Comment:
1. Move the test after `EnsureRequirements adds sort when required ordering isn't a prefix of existing ordering`?
2. Could we change the test to something like the following?
```scala
    val plan1 = DummySparkPlan(outputOrdering = Seq(orderingA, orderingB),
      outputPartitioning = HashPartitioning(exprA :: Nil, 5))
    val plan2 = DummySparkPlan(outputOrdering = Seq(orderingB, orderingC, orderingA),
      outputPartitioning = HashPartitioning(exprB :: Nil, 5))

    Seq(Inner ... ).foreach { joinType =>
      val fullSmj = SortMergeJoinExec(exprA :: Nil, exprB :: Nil, joinType, None, plan1, plan2)
      // assert
    }
```

Reply:
Done